spec/glueby/fee_provider/tasks_spec.rb

```diff
@@ -0,0 +1,506 @@
+            subject
+          end
+        end
+      end
+
+      context 'The wallet has no TPC' do
+        it 'doesn\'t broadcast tx'  do
+          expect_any_instance_of(Glueby::FeeProvider::Tasks).to receive(:status)
+          expect(wallet).not_to receive(:broadcast)
+          subject
+        end
+      end
+
+      context 'The wallet has some money but it is not enough to fill the UTXO pool' do
+        let(:unspents) do
+          [{
+            txid: '5c3d79041ff4974282b8ab72517d2ef15d8b6273cb80a01077145afb3d5e7cc5',
+            script_pubkey: '76a914234113b860822e68f9715d1957af28b8f5117ee288ac',
+            vout: 0,
+            amount: 10_100,
+            finalized: true
+          }]
+        end
+        let(:balance) { 10_100 }
+
+        it 'creates 9 fee outputs' do
+          expect(wallet).to receive(:broadcast) do |tx|
+            expect(tx.outputs[0...9].map(&:value).uniq).to eq([1_000])
+            # The change output's value should equal to <before amount> - <UTXO pool size> * <fixed fee> - <fee of the tx>
+            expect(tx.outputs[9].value).to eq(10_100 - 9 * 1_000 - 1_000)
+          end
+          expect(tasks).to receive(:status)
+          subject
+        end
+      end
+    end
+
+    context 'The UTXO pool has some outputs but it is not full' do
+      let(:pool_outputs) do
+        3.times.map do |i|
+          {
+            txid: 'f14b29639483da7c8d17b7b7515da4ff78b91b4b89434e7988ab1bc21ab41377',
+            vout: i,
+            script_pubkey: '21c2dbbebb191128de429084246fa3215f7ccc36d6abde62984eb5a42b1f2253a016bc76a914fc688c091d91789ccda7a27bd8d88be9ae4af58e88ac',
+            amount: 1_000,
+            finalized: true
+          }
+        end
+      end
+
+      context 'The wallet has enough TPC amount' do
+        let(:unspents) do
+          [{
+            txid: '5c3d79041ff4974282b8ab72517d2ef15d8b6273cb80a01077145afb3d5e7cc5',
+            script_pubkey: '76a914234113b860822e68f9715d1957af28b8f5117ee288ac',
+            vout: 0,
+            amount: 100_000_000,
+            finalized: true
+          }] + pool_outputs
+        end
+        let(:balance) { 100_003_000 }
+
+        it 'fills the UTXO pool' do
+          expect(wallet).to receive(:broadcast) do |tx|
+            expect(tx.inputs.count).to eq(1)
+            expect(tx.inputs[0].out_point).to eq(Tapyrus::OutPoint.from_txid('5c3d79041ff4974282b8ab72517d2ef15d8b6273cb80a01077145afb3d5e7cc5', 0))
+
+            # outputs count should be default UTXO pool size - existing fee output + a change output
+            expect(tx.outputs.count).to eq(18)
+            expect(tx.outputs[0...17].map(&:value).uniq).to eq([1_000])
+            # The change output's value should equal to <before amount> - <UTXO pool size> * <fixed fee> - <fee of the tx>
+            expect(tx.outputs[17].value).to eq(100_000_000 - 17 * 1_000 - 1_000)
+          end
+          expect(tasks).to receive(:status)
+          subject
+        end
+      end
+
```

Comment:
Please add the case where there are multiple inputs.

Reply:
done